### ISSUE TITLE
HexaPDF::FilterError (Problem while decoding Flate encoded stream: buffer error)

### ISSUE THREAD

**Issue by parradox2nd** [gettalong/hexapdf#68]
While trying to process a PDF, I got the error HexaPDF::FilterError (Problem while decoding Flate encoded stream: buffer error)

I used the code snippet below:
```
class SampleProcessor < HexaPDF::Content::Processor
  attr_reader :text
  def initialize()
    super()
    @text = ['']
  end

  def show_text(str)
    @text << decode_text(str)
  end
  alias :show_text_with_positioning :show_text
end
```

I found that when I performed the following:
```
reader = HexaPDF::Document.new(io: io)
  processor = SampleProcessor.new
  reader.pages.each do |page|
    page.process_contents(processor)
  end
```
The error would manifest.  The PDF file I used is attached.  My apologies if this is something easily addressed.  I do not have a deep understanding of the gem.
[OriginationDocuments-20190514.pdf](https://github.com/gettalong/hexapdf/files/3182705/OriginationDocuments-20190514.pdf)

**Comment by gettalong** [comment]
I found a problem with your PDF file (concerning the padding of an AES encrypted string), still have to determine whether HexaPDF is at fault or the document is faulty. Will keep you posted.

**Comment by gettalong** [comment]
Thanks for providing the faulty PDF file and sample code which helped a lot in figuring out the problems.

So, the problem with the AES encrypted string is really a problem with the file itself. However, I have still implemented a work-around for this kind of problem to make HexaPDF more resilient.

I also found the problem with the Flate encoded stream that is the root cause of your error message. This is also an error with the file itself because an empty stream was encoded incorrectly. I have implemented another work-around for this problem.

Will release a bug fix version with the fixes later!

**Comment by parradox2nd** [comment]
Thank you.  I will also pass along the file generation issue.

**Comment by gettalong** [comment]
@parradox2nd I have released the bug fix version 0.9.2.

**Comment by parradox2nd** [comment]
> @parradox2nd I have released the bug fix version 0.9.2.

Thank you Mr. Leitner!  Your fix worked.  I really appreciate your help.
